src/PrestaShopBundle/Controller/Admin/CommonController.php

```diff
@@ -207,4 +208,20 @@ public function renderSidebarAction($url, $title = '', $footer = '')
             'url' => urldecode($url),
         ]);
     }
+
+    /**
+     * @param $controller
+     * @param $action
+     * @return \Symfony\Component\HttpFoundation\RedirectResponse
+     * @throws \LogicException
+     */
+    public function resetSearchAction($controller, $action)
+    {
+        $employeeId = $this->getUser()->getId();
+        $shopId = $this->getContext()->shop->id;
+
+        $this->get('prestashop.core.admin.admin_filter.repository')->removeByEmployeeAndRouteParams($employeeId, $shopId, $controller, $action);
+
+        return new Response();
```

Comment:
Should be a redirect?

Reply:
it was at first step but it's useless because it's an AJAX call and I don't expect a response.

Redirect to another route/url will make the "server" works for no reason, isn't it?

Reply:
If it's ajax use `JsonResponse` instead which is recommended for ajax =)

Reply:
yes, and no: reset_search code expects html for now.

Reply:
oO weird